tests/test_default_gaussian.py

```diff
@@ -567,8 +567,8 @@ def test_unsupported_gates(self):
         for g in all_gates - gates:
             op = getattr(qml.ops, g)
 
-            if op.num_wires == 0:
-                wires = [0]
+            if op.num_wires <= 0:

```

Comment:
Can you kindly explain this test, specifically  the `<= 0 ` ? should we not check if `op.num_wires` is either 0 or -1 ?

Reply:
Ah, this is a test that has been deprecated and due to be refactored using PyTest :slightly_smiling_face: It's a pretty bad test - it loops over all unsupported/supported gates, and automatically applies them to make sure there is no failures.

This is pretty bad for two reasons:

1. The test is hard to maintain; adding operations will cause the existing test to fail. Ideally, adding a new operation should correspond to adding a new test, not fixing a large and cumbersome existing test.

2. Tests should not contain loops over test data. Each test function should be short, self contained, and explicit, ideally testing only a single thing.

Since we are planning to majorly refactor this test function ASAP, and this test didn't already take into account gates that apply to any number of wires, we should make an issue to remind ourselves to do this during the refactor.